### PR TITLE
init --analyze should return unique image names

### LINKED ISSUE #3128: Skaffold init --analyze returns duplicate image name
### Expected behavior
"skaffold init --analyze" should dedup image names. Expected output below.

{"dockerfiles":["src\\backend\\Dockerfile","src\\frontend\\Dockerfile"],"images":["node-guestbook-backend","mongo","node-guestbook-frontend"]}

### Actual behavior
"skaffold init --analyze" returns the "mongo" name twice in the "images". Actual output below.

{"dockerfiles":["src\\backend\\Dockerfile","src\\frontend\\Dockerfile"],"images":["node-guestbook-backend","mongo","node-guestbook-frontend","mongo"]}

### Information

- Skaffold version: version... v0.39.0-215-g96e5cef6
- Operating system: ... Windows
- Contents of skaffold.yaml:
No skaffold.yaml

```yaml
<paste your skaffold.yaml here>
```

### Steps to reproduce the behavior
1. Get the nodejs guestbook cloud code sample.
2. Delete the skaffold.yaml at the root
3. run "skaffold init --analyze"

### PR DESCRIPTION
**Pull request by caulagi**
Fixes #3128

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Mentions any output changes.
- [ ] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [ ] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

```
Bug fixes
    * "Skaffold init --analyze returns a unique list of image names"
```

### CODE REVIEW

File: pkg/skaffold/initializer/init.go [hunk 1]
```diff
@@ -544,7 +560,10 @@ func printAnalyzeJSON(out io.Writer, skipBuild bool, pairs []builderImagePair, u
 		a.Builders = append(a.Builders, Builder{Name: config.Name(), Payload: config})
 	}
 	for _, image := range unresolvedImages {
-		a.Images = append(a.Images, Image{Name: image, FoundMatch: false})
+		if _, ok := seen[image]; !ok {
```
Comment (dgageot) [inline pkg/skaffold/initializer/init.go]: `if !seen[image]`

File: pkg/skaffold/initializer/init.go [hunk 2]
```diff
@@ -462,6 +462,20 @@ func canonicalizeName(name string) string {
 	return canonicalized[:253]
 }
 
+// deduplicate removes any duplicated values and returns a new slice, keeping the order unchanged
+func deduplicate(s []string) []string {
+	encountered := map[string]bool{}
+	ret := make([]string, 0)
```
Comment (dgageot) [inline pkg/skaffold/initializer/init.go]: `var ret []string`

**Comment by caulagi** [comment]
@balopat, @dgageot Thanks for the feedback. Can you have one more look?

**Comment by caulagi** [comment]
Oops, fixed.

**Pull request merged**